desc/magnetic_fields.py

```diff
@@ -1271,18 +1342,89 @@ def field_line_integrate(
     x0 = jnp.array([r0, phis[0] * jnp.ones_like(r0), z0]).T
 
     @jit
-    def odefun(rpz, s):
+    def odefun(s, rpz, args):
         rpz = rpz.reshape((3, -1)).T
         r = rpz[:, 0]
         br, bp, bz = field.compute_magnetic_field(rpz, params, basis="rpz", grid=grid).T
         return jnp.array(
             [r * br / bp * jnp.sign(bp), jnp.sign(bp), r * bz / bp * jnp.sign(bp)]
         ).squeeze()
 
-    intfun = lambda x: odeint(odefun, x, phis, rtol=rtol, atol=atol, mxstep=maxstep)
+    # diffrax parameters
+    stepsize_controller = PIDController(rtol=rtol, atol=atol, dtmin=min_step_size)
```

Comment:
might be better to make this a user supplied input with default? to allow for other controllers, constant step size etc.